biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx

```diff
@@ -39,35 +95,31 @@ export const SimulatePage = () => {
     history.push('/');
   };
 
-  const stateToGridItemRet = (state: any): GridItemRet[] => {
-    return state.nodes.map((node: any) => {
-      return {
-        gridItemName: node.value.gridItemName,
-        relativePosition: node.value.position ||
-          node.value.relativePosition || { x: 0, y: 0 },
-      };
-    });
-  };
-  const stateToGridItemLines = (state: any): GridItemLines[] => {
-    return state.edges.map((edge: any) => {
-      return {
-        fromItem: edge.v,
-        toItem: edge.w,
-        // TODO this is not a correct value
-        powerThroughLinesKiloWatts: edge.value.power,
-      };
-    });
+  const pauseSimulation = async () => {
+    controlSimulation?.pauseFN();
   };
 
-  const playSimulation = () => {
+  const play = () => {
     const simResultsStateLen = simulationResults?.states.length || 0;
-    if (currentStateFrame >= simResultsStateLen) {
-      return;
-    }
-    setTimeout(() => {
-      setCurrentStateFrame(currentStateFrame + 1);
-      playSimulation();
-    }, 1000);
+    let finished = false;
+    let pause = () => {
+      finished = true;
+    };
+    const runThroughSteps = async () => {
+      for (let i = stateFrame; i < simResultsStateLen; i++) {
+        if (finished) {
+          return;
+        }
+        await new Promise((res, rej) =>
```

Comment:
it needs a promise so that the setTimeout function can be awaited to be called. So, the promise only resolves after the second of waiting is up. Thus the state frames are stepped through every second